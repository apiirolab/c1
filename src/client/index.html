<!DOCTYPE html>
<html>

<head>
  <title></title>
  <meta name="description" content="Corporate process management for Western Sydney University!">
  <link rel="icon" type="image/png" href="/favico.png">
  <link href='https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.3.2/semantic.min.css' rel='stylesheet' type='text/css'>
</head>

<body>
  <div id="root"></div>
  $bundles
</body>

</html>