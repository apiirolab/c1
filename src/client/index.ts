export const info = 'Client Info';
