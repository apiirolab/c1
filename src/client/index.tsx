import * as React from 'react';
import * as ReactDOM from 'react-dom';

import { ApolloProvider } from 'react-apollo';
import { Routes } from './config/router';
import { client } from './config/apollo';
// import * as serviceWorker from './registerServiceWorker';

ReactDOM.render(
  <ApolloProvider client={client}>
    <Routes />
  </ApolloProvider>,
  document.getElementById('root')
);

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: http://bit.ly/CRA-PWA
// serviceWorker.unregister();
