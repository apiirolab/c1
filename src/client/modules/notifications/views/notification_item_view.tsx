import * as dayjs from 'dayjs';
import * as React from 'react';


// import relativeTime from 'dayjs/plugin/relativeTime'
import { List } from 'semantic-ui-react';
import styled from 'styled-components';

// dayjs.extend(relativeTime);

const Time = styled.span`
  color: #aaa;
`;

type Props = {
  notification: Corpix.Entities.Notification;
};

export const NotificationView: React.SFC<Props> = ({ notification }) => (
  <List.Item>
    <Time>{ dayjs(notification.date).toString() }</Time> &middot; { notification.comment }
  </List.Item>
);
