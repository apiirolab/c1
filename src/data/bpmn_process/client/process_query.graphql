query ProcessQuery($id: ID!) {
  process(id: $id) {
    actionCount
    description
    id
    model
    resources {
      name
      type
    }
    status
    version
  }
}
