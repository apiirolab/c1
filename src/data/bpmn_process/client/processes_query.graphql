query Processes($input: BpmnProcessesInput!) {
  processes(input: $input) {
    actionCount
    description
    id
    name
    type
  }
}
