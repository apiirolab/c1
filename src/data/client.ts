import * as QueryTypes from './generated/types';

export { default as gql } from 'graphql-tag';

export { QueryTypes };

export function purge<T>(params: any): T {
  for (let key of Object.getOwnPropertyNames(params)) {
    if (!params[key]) {
      delete params[key];
    }
    if (params[key] instanceof Object) {
      purge(params[key]);
    }
  }
  return params;
}
