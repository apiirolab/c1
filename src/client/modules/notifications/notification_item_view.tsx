import * as React from 'react';

import { QueryTypes } from 'data/client';

import { Link } from '@reach/router';
import { inject } from 'mobx-react';
import { Icon, List } from 'semantic-ui-react';
import styled from 'styled-components';

import { RemoveNotificationMutation } from './notifications_queries';

// dayjs.extend(relativeTime);

const Time = styled.span`
  color: #aaa;
`;

const ClickableIcon = styled(Icon)`
  cursor: pointer;
`;

type Props = {
  notification: QueryTypes.Notifications;
  context?: App.Context;
};

export const NotificationView = inject('context')(({ notification, context }: Props) => (
  <List.Item>
    <List.Icon>
      <RemoveNotificationMutation
        optimisticResponse={{
          removeNotification: notification.id
        }}
      >
        {(removeNotification, { loading }) =>
          loading ? (
            <Icon name="hourglass half" color="grey" title={context.i18n`Removing`} />
          ) : (
            <ClickableIcon
              onClick={() => removeNotification({ variables: { id: notification.id } })}
              name="eye"
              color="grey"
              title={context.i18n`Mark as viewed`}
            />
          )
        }
      </RemoveNotificationMutation>
    </List.Icon>
    <List.Content>
      <div key="1">
        <Time>{context.Ui.relativeDate(notification.createdAt)}</Time> &middot;{' '}
        <Link to={`/process/${context.Ui.urlName(notification.processInstance.process.name)}`}>
          {notification.text}
        </Link>
      </div>
    </List.Content>
  </List.Item>
));
