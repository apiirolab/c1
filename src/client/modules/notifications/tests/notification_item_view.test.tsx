import * as React from 'react';

import { create, MockedProvider, QueryTypes, render } from 'client/tests';
import { List } from 'semantic-ui-react';
import { NotificationView } from '../notification_item_view';
import { createNotification } from './notifications_test_data';

describe('Notifications', () => {
  describe('Item', () => {
    function componentWithData(notification: QueryTypes.Notifications = createNotification()) {
      return (
        <MockedProvider>
          <NotificationView notification={notification} context={create.context()} />
        </MockedProvider>
      );
    }
    it('renders default view', () => {
      const root = render(componentWithData(createNotification()));
      expect(root).toMatchSnapshot();
    });

    return {
      component: <List>{componentWithData()}</List>
    };
  });
});
