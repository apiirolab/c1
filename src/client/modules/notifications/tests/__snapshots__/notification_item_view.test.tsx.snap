// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications Item renders default view 1`] = `
<div
  className="item"
  onClick={[Function]}
  role="listitem"
>
  <i
    aria-hidden="true"
    className="icon"
  >
    <i
      aria-hidden="true"
      className="grey eye icon sc-bwzfXH kZUMPF"
      onClick={[Function]}
      title="Mark as viewed"
    />
  </i>
  <div
    className="content"
  >
    <div>
      <span
        className="sc-bdVaJa cWCCxy"
      >
        2 months ago
      </span>
       Â·
       
      <a
        href="/process/process-name"
        onClick={[Function]}
      >
        Notification Text
      </a>
    </div>
  </div>
</div>
`;
